File: android/blemodule/src/main/java/net/qaul/ble/model/BLEScanDevice.kt
package net.qaul.ble.model

import android.bluetooth.BluetoothDevice
import android.bluetooth.le.ScanResult

abstract class BLEScanDevice  {
    companion object {
        fun getDevice(): BLEScanDevice {
            return BLEDevice()
        }
    }
    abstract var deviceRSSI: Int
    abstract var scanResult: ScanResult?
    abstract var bluetoothDevice: BluetoothDevice?
    abstract var name: String?
    abstract var macAddress: String?
    abstract var isConnectable: Boolean
}

File: iOS/QaulBLE/QaulBLE/iOS App/Model/BLEScanDevice.swift
//
//  BLEScanDevice.swift
//  QaulBLE
//
//  Created by BAPS on 31/01/22.
//

import UIKit

public class BLEScanDevice: NSObject {
    
    var deviceRSSI: Int = 0
    
//    var scanResult: ScanResult? skip for iOS
    var name: String?
    var macAddress: String? //UUID
    var intervalNanos: Double?
    var bluetoothDevice: BLEDevice?
    var isConnectable = true
    var lastFoundTime: Double?
    var qaulId: Data?
    var strqaulId: String?
    
    public override init() {
        
    }
}


